great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -52,69 +54,63 @@ class SQLDatasourceWarning(UserWarning):
     pass
 
 
-@pydantic_dc.dataclass(frozen=True)
-class ColumnSplitter:
+class _ColumnSplitter(ExperimentalBaseModel):
     column_name: str
     method_name: str
 
+    @property
+    def param_names(self) -> List[str]:
+        raise NotImplementedError
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
+        raise NotImplementedError
+
+    def request_options_to_batch_spec_kwarg_identifiers(
+        self, options: BatchRequestOptions
```

Comment:
@billdirks Not for immediately now, but in a near future, I am thinking of putting the sorting for files into DataConnector (as it was in V3).  In the DataConnector interface set, there is a method that converts `BatchDefition` to `BatchSpec`, and a private method that it uses that does exactly what this one here.  I will investigate at an opportune time if there are any commonalities and bring up any ideas for consideration (no action for now, just thinking ahead).  Thanks.